### PR TITLE
Modify 'web suite' link to be a dropdown menu containing apps

### PR DESCRIPTION
**Pull request by VoxSecundus**
This PR introduces a change to the landing page navbar. The conditionally used `Web Suite` button is now a split dropdown button. The old button functionality remains the same, but it is now followed by a dropdown button listing the available webapps provided to `flight-webapp-components` by the webapp currently being visited.

### CODE REVIEW

**Comment by benarmston** [comment]
@VoxSecundus webapp components (and our webapps in general) are already using [reactstrap](https://deploy-preview-2356--reactstrap.netlify.app/components/button-dropdown/) as the react bootstrap library.  Can you remove the use of `react-bootstrap` and change this to use `reactstrap` please.

**Comment by VoxSecundus** [comment]
@benarmston I've just updated the branch to use `reactstrap` instead of `react-bootstrap`.

**Comment by benarmston** [comment]
@VoxSecundus can you rebease this on `origin/master` and then I'll take another look.

```sh
ben@midgard:~/projects/flight-webapp-components (enh/web-suite-dropdown *$% u=)
$ git log  --oneline origin/master --not HEAD
5ee4ab0 (tag: 0.3.0, origin/master, origin/HEAD, master) Merge branch 'release/0.3.0'
3e3c71b Remove old test artefacts
4dc83f4 Bump version to 0.3.0
c2ce982 `./bin/release.sh` learns `--version NEW_VERSION`
c9ba386 Bump .nvmrc
f1937eb Merge pull request #5 from openflighthpc/enh/sign-in-modal-copy
491fa14 Build artefacts
e3e22ac Better form customization
1fe7a14 Add support for importing sign in modal text from branding
6eb0790 Merge pull request #6 from openflighthpc/enh/fold-in-wae-changes
215c584 Build artefacts
e3186b8 Update Footer to have configurable copyright
9e1e0dc Support additional items on brandbar
b4d5498 Fold in WAE branding changes
```

(You'll likely need to do the same for the UMD login widget work).

**Comment by VoxSecundus** [comment]
Rebased on `master` successfully

File: src/BrandBar.js [hunk 1]
```diff
@@ -33,16 +34,40 @@ function BrandBarItems({ className }) {
...
+	       <span className={"fa fa-solid fa-fw fa-" + app.fa_icon}></span>&nbsp;
+	       {app.short_title || app.title}
+	     </DropdownItem>
+  });
+
+  const [dropdownOpen, setOpen] = React.useState(false);
+  const toggle = () => setOpen(!dropdownOpen);
+
   const appsLink = (
     <li className="nav-item">
```
Comment (benarmston) [inline src/BrandBar.js]: Can you add a `dropdown` class to this please?

**Review by benarmston — CHANGES_REQUESTED** [review]
This is close.  A `dropdown` class is missing and also some font colour issues with the alces branding.

image

**Comment by VoxSecundus** [comment]
@benarmston I'm not getting the font colour issue on my build; have you made any local changes to the `alces-branding` type?

File: src/brandbar.module.css [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+.dropdownItem:hover {
+  color: black;
+}
+
+.dropdownToggle {
+  display: flex;
+  align-items: center;
+}
```
Comment (benarmston) [inline src/brandbar.module.css]: This shouldn't be necessary.  If you have the webapp's loading their styles from the dev landing page, they should already be picking up this setting from `branding.css`.

File: src/brandbar.module.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.dropdownItem {
+  padding: 8px;
```
Comment (benarmston) [inline src/brandbar.module.css]: This padding isn't wanted.

The landing page `dropdown-item`s incorrectly have the `nav-link nav-menu-button` classes.  The `nav-link` class adds some unwanted `padding: 0.5rem`.  We want the default padding for `.dropdown-item`.  (Or something similar to that.  Either way we should be using the default padding for `.dropdown-item`).

The fix to this is

1. Remove `nav-link nav-menu-button` from `dropdown-item` in flight landing page repo.
2. Remove this padding here.

File: src/brandbar.module.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  color: white;
+}
+
+.dropdownItem:hover {
+  color: black;
+}
```
Comment (benarmston) [inline src/brandbar.module.css]: These colours aren't going to work with the default (openflight) branding.  Colours such as this should be specified in the `branding.css` files in landing page and picked up by the webapp.

Remove these colours and make the changes to branding.css I've asked for on flight-landing-page.

**Review by benarmston — CHANGES_REQUESTED** [review]
After following through the flight direct branding dropdown item colour issues, I have some more comments on this PR.  These comments go hand-in-hand with those I've added to https://github.com/openflighthpc/flight-landing-page/pull/16

**Comment by benarmston** [comment]
LGTM.

**Pull request merged**